yt/data_objects/data_containers.py

```diff
@@ -2179,3 +2182,93 @@ def _reconstruct_object(*args, **kwargs):
     obj = cls(*new_args)
     obj.field_parameters.update(field_parameters)
     return ReconstructedObject((ds, obj))
+
+def monte_carlo_sample(ds, n_samples=100000, fields=None, n_neighbors=1):
+    """
+    Uses Monte Carlo sampling to create a particle-based dataset out of a
+    grid-based dataset.  For each sample, the code probabilistically selects
+    a cell weighted according to its mass, and a particle is created possessing
+    the gas fields of the cell at some random location within the cell's volume.
+
+    By default the fields that are created are mass, density, smoothing_length,
+    and the position and velocity fields.
+
+    Parameters
+    ----------
+    ds : Dataset
+        Grid-based dataset used as source
+
+    n_samples : int
+        Number of times to sample the dataset
+
+    fields : list
+        List of additional fields you want to include in returned dataset
+
+    n_neighbors : int
+        Number of neighbors for each node in kdtree to determine
+        smoothing_length
+
+    Returns
+    -------
+    StreamParticlesDataset
+
+    Examples
+    --------
+    >>> ds_grid = yt.load('IsolatedGalaxy/galaxy0030/galaxy0030')
+    >>> ds_sph = monte_carlo_sample(ds_grid)
+    """
+    from yt.frontends.stream.api import \
+        load_particles
+    l_unit = 'code_length'
+    m_unit = 'code_mass'
+    d_unit = 'code_mass/code_length**3'
+    v_unit = 'code_length/code_time'
+    rands = np.random.rand(n_samples)
+    m_tot = ds.r[('gas', 'cell_mass')].sum()
+    bins = np.cumsum(ds.r[('gas', 'cell_mass')] / m_tot)
+    indices = np.digitize(rands, bins)
+    x_off = np.random.uniform(-0.5, 0.5, n_samples)
```

Comment:
do you anticipate anyone ever wanting to change the random function?

Reply:
I didn't really, no.  I guess i could make this a kwarg, but given the nature of Eulerian hydrodynamics, it seems appropriate to uniformly sample the entirety of a cell's volume as opposed to weighting it toward the center in any way.  But I could make this more flexible.

Reply:
Ah!  Makes sense.  I think you've answered my question and it doesn't need any changing!